pkg/apis/monitoring/v1alpha1/types.go

```diff
@@ -55,8 +55,10 @@ type MonitoringStackSpec struct {
 	LogLevel LogLevel `json:"logLevel,omitempty"`
 
 	// Label selector for Monitoring Stack Resources.
-	// +optional
-	ResourceSelector *metav1.LabelSelector `json:"resourceSelector,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default={}
```

Comment:
`+kubebuilder:default` doesn't seem to play well with pointer fields: the default value doesn't appear in the generated YAML manifest of the CRD :thinking:

Reply:
I suspect that kubebuilder's defaulting works well with value fields but not so much with pointers.